### PR TITLE
[RFC] Truncate system.*_log tables before each test

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Non-significant (changelog entry is not required)

Some tests uses `system.*_log` tables to verify some information, and they have `event_time`/`event_date` filter, which looks icky, by truncate the table before each test, the code can became cleaner and more sane (since some tests check for some seconds, say 300, and you cannot run this test in the loop).

I've made it by default, since looks like a sane default, and because otherwise docker images should be updated (and I prefer not to bother anyone), so comments are welcome!

P.S. another way to avoid this is by creating temporary materialized view for the table and use it for querying, but this will require some extra queries.

### CODE REVIEW

File: tests/clickhouse-test [hunk 1]
```diff
@@ -234,6 +234,11 @@ def run_tests_array(all_tests_with_params):
                         clickhouse_proc = Popen(shlex.split(args.client), stdin=PIPE, stdout=PIPE, stderr=PIPE)
                         clickhouse_proc.communicate("SELECT 'Running test {suite}/{case} from pid={pid}';".format(pid = os.getpid(), case = case, suite = suite))
 
+                    if not args.no_system_log_cleanup:
+                        for table in ['query_log', 'query_thread_log', 'trace_log', 'metric_log']:
+                            clickhouse_proc = Popen(shlex.split(args.client), stdin=PIPE, stdout=PIPE, stderr=PIPE)
+                            clickhouse_proc.communicate("TRUNCATE TABLE IF EXISTS system.{}".format(table))
```
Comment (filimonov) [inline tests/clickhouse-test:240]: I think you also should do `system flush logs` before. Otherwise it can write queued logs just after truncation.

Comment (azat) [reply tests/clickhouse-test:240]: Nice catch!

**Review by alexey-milovidov — APPROVED** [review]
Yes, that's what we need.

**Comment by azat** [comment]
Please note that I've also disabled `system_tables_lazy_load` (description updated accordingly)

And this requires docker image update (although tests will pass even without updating the container, since none of existing tests in upstream/master depends on this).

**Comment by alexey-milovidov** [comment]
Is there any reasons why we shouldn't disable `system_tables_lazy_load` by default?

**Comment by azat** [comment]
>Is there any reasons why we shouldn't disable system_tables_lazy_load by default?

That was my [initial suggestion](https://github.com/ClickHouse/ClickHouse/pull/9642#discussion_r401421174) :)

**Comment by alexey-milovidov** [comment]
Let's do it...

**Comment by azat** [comment]
>Let's do it...

Ok, will do separately (and I removed patch with disable it for tests separately)

**Comment by azat** [comment]
>Functional stateless tests (release)

was ~28 min, now 32 min (may be time related, but not sure)

**Comment by alexey-milovidov** [comment]
It's Ok.

**Pull request merged**